yt_dlp/extractor/maariv.py

```diff
@@ -0,0 +1,70 @@
+from .common import InfoExtractor
+import re
+
+
+class MaarivIE(InfoExtractor):
+    IE_NAME = 'maariv.co.il'
+    _VALID_URL = r'(https?://)?(www\.)?(m\.)?maariv\.co\.il(?:/[^/]+)*/Article-(?P<id>\d{7})'
+    _TESTS = [
+        {
+            'url': 'https://www.maariv.co.il/news/law/Article-1044008',
+            'info_dict': {
+                'id': '1044008',
+                'title': '.*',
+            },
+        }
+    ]
+
+    @staticmethod
+    def extract_resolution(url):
+        match = re.search(r'(\d{2,4}x\d{2,4})\.mp4$', url)
+        return match.group(1) if match else None
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        # Find the correct iframes
+        video_urls = re.findall(r'<iframe [^>]?poster="([^"]+)"[^>]+ src="([^"]+)"', webpage)
+        video_info_list = []
+
+        for thumbnail, src in video_urls:
+            media_param = re.search(r'media=(\d+)', src).group(1)
+            info_json = self._download_json(f"https://dal.walla.co.il/media/{media_param}?origin=player.maariv.co.il",
+                                            video_id)
+            data = info_json['data']
+            main_video_to_download_filename = data['video']['file_name']
+            main_video_to_download_url = data['video']['url']
+            duration_seconds = int(data['video']['duration'])
+
+            format_list = [
+                {
+                    'format_id': '0',
+                    'url': main_video_to_download_url,
+                    'resolution': 'Main Video',
+                }
+            ]
+
+            for format_id, stream_url_object in enumerate(data['video']['stream_urls'], start=1):
+                stream_url = stream_url_object['stream_url']
+                format_list.append({
+                    'format_id': str(format_id),
+                    'url': stream_url,
+                    'resolution': self.extract_resolution(stream_url),
+                })
```

Comment:
We can also make the format extraction non-fatal and make use of some helper functions
```suggestion

            formats = []
            hls_url = traverse_obj(data, ('video', 'url', {url_or_none}))
            if hls_url:
                formats.extend(self._extract_m3u8_formats(hls_url, video_id, m3u8_id='hls', fatal=False))

            for http_format in traverse_obj(data, ('video', 'stream_urls', ..., 'stream_url', {url_or_none})):
                formats.append({
                    'url': http_format,
                    'format_id': 'http',
                    **parse_resolution(http_format),
                })
```